### PR TITLE
Snap methods to Injected

### LINKED ISSUE #53: Add snap methods to injected
- extend `Injected` interface with method `getMetamaskSnapApi` in which you will create and return following object:
```typescript
interface MetamaskSnapApi {
   addAsset(): Promise<void>
  ...rest of snap methods
}
```

### PR DESCRIPTION
**Pull request by irubido**
resolves #53

### CODE REVIEW

File: adapter/methods.ts [hunk 1]
```diff
@@ -43,4 +53,91 @@ export async function addKusamaAsset(): Promise<void> {
...
+    return await window.ethereum.send({
+        method: pluginOrigin,
+        params: [{
+            method: 'configure',
+            params: {configuration: configuration}
+        }]
+    });
+}
+
+export async function getAllTransactions(address?: string): Promise<any> {
```
Comment (mpetrunic) [inline adapter/methods.ts:129]: you should add return type to all responses, but I suggest you rebase of monorepo branch so you can add all interfaces and types in single place and share with other components

**Comment by irubido** [comment]
PR rebased and adjusted to Monorepo conversion branch

File: packages/adapter/src/methods.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+async function sendSnapMethod(method: 'getBalance' | 'getAddress' | 'getPublicKey' | 'exportSeed', pluginOrigin: string): Promise<string|null> {
+  try {
+      await window.ethereum.send({
+          method: pluginOrigin,
+          params: [{
+              method: method
+          }]
+      });
+  } catch (e) {
```
Comment (mpetrunic) [inline packages/adapter/src/methods.ts:74]: Don't catch this, error won't be propagated to user

File: packages/adapter/src/methods.ts [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+}
+
+export async function getBalance(pluginOrigin: string): Promise<string> {
+  const response = await sendSnapMethod("getBalance", pluginOrigin);
+  return (response != null) ? response : "Unable to fetch balance";
+}
+
+export async function getAddress(pluginOrigin: string): Promise<string> {
+  const response = await sendSnapMethod("getAddress", pluginOrigin);
+  return (response != null) ? response : "Unable to fetch address";
```
Comment (mpetrunic) [inline packages/adapter/src/methods.ts:87]: Didn't mean you to return string to user, just let error bubble up and user will catch it and handle it

File: packages/example/package.json [hunk 4]
```diff
@@ -15,6 +16,7 @@
     "@types/node": "^12.0.0",
     "@types/react": "^16.9.0",
     "@types/react-dom": "^16.9.0",
+    "polkadot-snap": "0.0.1",
```
Comment (mpetrunic) [inline packages/example/package.json]: I dont' think we need that as dependency, probably just @nodefactory/metamask-polkadot-adapter and @nodefactory/metamask-polkadot-types

File: packages/adapter/package.json [hunk 5]
```diff
@@ -17,7 +17,8 @@
     "build:types": "tsc --emitDeclarationOnly",
     "lint": "yarn run lint:style && yarn run lint:types",
     "lint:types": "tsc --noEmit --pretty",
-    "lint:style": "eslint --color 'src/**/*.{js,ts,tsx}'",
+    "lint:style": "eslint --color 'src/**/*.{js,ts,tsx}' --fix",
```
Comment (mpetrunic) [inline packages/adapter/package.json:20]:
```suggestion
    "lint:style": "eslint --color 'src/**/*.{js,ts,tsx}'",
```

File: packages/adapter/package.json [hunk 6]
```diff
@@ -17,7 +17,8 @@
...
+    "lint:fix": "yarn run lint --fix",
```
Comment (mpetrunic) [inline packages/adapter/package.json:21]:
```suggestion
    "lint:style:fix": "yarn run lint:style --fix",
```

File: packages/adapter/package.json (hunk 5 above)
Comment (mpetrunic) [reply packages/adapter/package.json:20]: Otherwwise it's gonna mess with CI

File: package.json [hunk 7]
```diff
@@ -14,6 +14,7 @@
     "build": "lerna run build",
     "test": "lerna run test",
     "lint": "lerna run lint",
+    "lint:fix": "lerna run lint:fix",
```
Comment (mpetrunic) [inline package.json:17]:
```suggestion
    "lint:style:fix": "lerna run lint:style:fix",
```

File: packages/adapter/src/methods.ts [hunk 8]
```diff
@@ -47,54 +47,51 @@ export async function isPolkadotSnapInstalled(pluginOrigin: string): Promise<boo
...
-      method: pluginOrigin,
-      params: [{
-          method: 'addKusamaAsset'
-      }]
+    method: pluginOrigin,
+    params: [{
+      method: 'addKusamaAsset'
+    }]
   });
 }
```
Comment (mpetrunic) [inline packages/adapter/src/methods.ts:55]: You should delete this whole function as it got replaced with `addPolkadotAsset`

File: packages/adapter/src/methods.ts [hunk 9]
```diff
@@ -45,6 +63,37 @@ export async function addAsset(pluginOrigin: string): Promise<void> {
...
 
+async function sendSnapMethod(method: 'getBalance' | 'getAddress' | 'getPublicKey' | 'exportSeed', 
+  pluginOrigin: string): Promise<string> {
+  const response = await window.ethereum.send({
+    method: pluginOrigin,
+    params: [{
+      method: method
+    }]
+  });
+  return response as string;
```
Comment (mpetrunic) [inline packages/adapter/src/methods.ts:74]: response isn't always of type string, should be unknown

File: packages/adapter/src/methods.ts [hunk 10]
```diff
@@ -55,4 +104,44 @@ export async function configure(pluginOrigin: string, config: SnapConfig): Promi
...
+      }]
+    });
+    return blockResponse as BlockInfo;
+  } catch (e) {
+    console.log("Unable to fetch latest block", e);
+    return {hash: "", number: ""};
+  }
+}
+
+export async function setConfiguration(configuration: SnapConfig, pluginOrigin: string): Promise<void> {
```
Comment (mpetrunic) [inline packages/adapter/src/methods.ts:127]: be consistent, origin should always be first param

File: packages/adapter/src/snap.ts [hunk 11]
```diff
@@ -49,8 +52,26 @@ export class MetamaskPolkadotSnap implements Injected {
...
+      exportSeed,
+      getAccountAddress,
+      getAddress,
+      getAllTransactions,
+      getBalance,
+      getLatestBlock,
+      getPublicKey,
+      hasMetaMask,
+      installPolkadotSnap,
+      isPolkadotSnapInstalled,
```
Comment (mpetrunic) [inline packages/adapter/src/snap.ts:73]: you don't need to export this method, same goes for `installPolkadotSnap`

File: packages/adapter/src/snap.ts [hunk 12]
```diff
@@ -49,8 +52,26 @@ export class MetamaskPolkadotSnap implements Injected {
...
       await configure(this.pluginOrigin, this.config);
-      await addAsset(this.pluginOrigin);
+      await addPolkadotAsset(this.pluginOrigin);
     }
     return this;
   };
+
+  public getMetamaskSnapApi = async (): Promise<MetamaskSnapApi> => {
+    return {
+      addKusamaAsset,
```
Comment (mpetrunic) [inline packages/adapter/src/snap.ts:62]: remove this

File: packages/example/src/containers/Dashboard/Dashboard.tsx [hunk 13]
```diff
@@ -32,13 +32,15 @@ export const Dashboard = () => {
     useEffect(() => {
         (async () => {
             if(state.polkadotSnap.isInstalled) {
-                await setConfiguration({networkName: network});
+                const x = await setConfiguration({networkName: network});
+                console.log(x);
```
Comment (mpetrunic) [inline packages/example/src/containers/Dashboard/Dashboard.tsx:36]: Don't commit your debugging statements

File: packages/types/index.d.ts [hunk 14]
```diff
@@ -41,6 +41,10 @@ export interface AddPolkadotAssetRequest {
     method: "addPolkadotAsset";
 }
 
+export interface AddKusamaAsset {
+    method: "addKusamaAsset";
+}
```
Comment (mpetrunic) [inline packages/types/index.d.ts:46]: Remove this

File: packages/adapter/src/methods.ts [hunk 15]
```diff
@@ -64,32 +55,32 @@ export async function addPolkadotAsset(pluginOrigin: string): Promise<void> {
 }
 
 async function sendSnapMethod(method: 'getBalance' | 'getAddress' | 'getPublicKey' | 'exportSeed', 
-  pluginOrigin: string): Promise<string> {
+  pluginOrigin: string): Promise<unknown> {
```
Comment (mpetrunic) [inline packages/adapter/src/methods.ts:58]: method: MetamaskPolkadotRpcRequest["method"]

File: packages/adapter/src/methods.ts [hunk 16]
```diff
@@ -64,32 +55,32 @@ export async function addPolkadotAsset(pluginOrigin: string): Promise<void> {
...
     params: [{
       method: method
     }]
   });
-  return response as string;
+  return response;
 }
 
-export async function getBalance(pluginOrigin: string): Promise<string> {
+export async function getBalance(pluginOrigin: string): Promise<unknown> {
```
Comment (mpetrunic) [inline packages/adapter/src/methods.ts:68]: Can you please type response as string. Same goes for below methods

File: packages/adapter/src/types.ts [hunk 17]
```diff
@@ -4,17 +4,12 @@ import {BlockInfo} from "../../../packages/snap/src/rpc/substrate/getBlock";
...
-  addKusamaAsset(pluginOrigin: string): Promise<void>;
   addPolkadotAsset(pluginOrigin: string): Promise<void>;
-  getBalance(pluginOrigin: string): Promise<string>;
-  getAddress(pluginOrigin: string): Promise<string>;
-  getPublicKey(pluginOrigin: string): Promise<string>;
-  exportSeed(pluginOrigin: string): Promise<string>;
-  hasMetaMask(pluginOrigin: string): boolean;
+  getBalance(pluginOrigin: string): Promise<unknown>;
+  getAddress(pluginOrigin: string): Promise<unknown>;
+  exportSeed(pluginOrigin: string): Promise<unknown>;
```
Comment (mpetrunic) [inline packages/adapter/src/types.ts:10]: why unknown? You can typehint what responses are expected

File: packages/adapter/src/methods.ts (hunk 15 above)
Comment (mpetrunic) [reply packages/adapter/src/methods.ts:59]: You are missing ["method"] @irubido

**Pull request merged**